pkg/capabilities/validate.go

```diff
@@ -0,0 +1,210 @@
+package capabilities
+
+import (
+	"encoding/json"
+	"errors"
+	"reflect"
+
+	"github.com/invopop/jsonschema"
+	"github.com/mitchellh/mapstructure"
+	jsonvalidate "github.com/santhosh-tekuri/jsonschema/v5"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+)
+
+type Validator[Config any, Input any, Output any] struct {
+	ValidatorArgs
+
+	configType Config
+	inputType  Input
+	outputType Output
+
+	schemas map[string]string
+}
+
+var uriPrefix = "https://github.com/smartcontractkit/chainlink/capabilities/"
+
+var _ Validatable = (*Validator[any, any, any])(nil)
+
+type ValidatorArgs struct {
+	Info CapabilityInfo
```

Comment:
@HenryNguyen5 It would be nice to not depend on CapabilityInfo if we can here. It looks like we only need it for the ID, is that correct? If so, maybe we can just pass in the ID?

Reply:
We use `ID` and `Description`. I figured it was good to pass in the entire struct since we can store a lot of useful metadata in the schema.